pkgs/applications/misc/p2pool/default.nix

```
@@ -0,0 +1,39 @@
+{ stdenv
+, cmake
+, fetchFromGitHub
+, gss
+, hwloc
+, lib
+, libsodium
+, libuv
+, openssl
+, pkg-config
+, zeromq
+}:
+
+stdenv.mkDerivation rec {
+  pname = "p2pool";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "SChernykh";
+    repo = "p2pool";
+    rev = "v${version}";
+    sha256 = "sha256-UyHyb3GXdvd5cDRAejLvHQO5bbcLFdZDWohZm2B6PLA=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ libuv zeromq libsodium gss hwloc openssl ];
+
+  installPhase = ''
+    install -vD p2pool $out/bin/p2pool
+  '';
```

It's inconvenient, but you need to run the pre/post hooks when overriding a phase:
```suggestion
  installPhase = ''
    runHook preInstall
    install -vD p2pool $out/bin/p2pool
    runHook preInstall
  '';
```

---

May you explain why it's necessary? I did this based on the `xmrig` package.

---

Without doing this if someone else adds a `preInstall` or `postInstall`, this will not work because since you have overridden `installPhase` `preInstall` and `postInstall` will not be called automatically anymore.

---

Even if this derivation itself doesn't use it, maybe someone will try to do `overrideAttrs` and just be confused when it doesn't work as expected.

---

I see now. Nice to know this, thanks!